test/jdk/com/sun/jdi/SuspendAfterDeath.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8287847
+ * @summary Test suspending a thread after it has terminated.
+ * @enablePreview
+ * @requires vm.continuations
+ * @run build TestScaffold VMConnection TargetListener TargetAdapter
+ * @run compile SuspendAfterDeath.java
+ * @run main/othervm SuspendAfterDeath
+ */
+import com.sun.jdi.*;
+import com.sun.jdi.event.*;
+import com.sun.jdi.request.*;
+import java.util.*;
+
+class SuspendAfterDeathTarg {
+    static final String THREAD_NAME = "duke";
+
+    // breakpoint here
+    static void done() {
+    }
+
+    public static void main(String[] args) throws Exception {
+        boolean useVirtualThread = ((args.length > 0) && args[0].equals("Virtual"));
+        Thread thread;
+        System.out.println("Starting debuggee " + (useVirtualThread ? "virtual" : "platform") + " thread.");
+        if (useVirtualThread) {
+            thread = Thread.ofVirtual().name(THREAD_NAME).start(() -> { });
+        } else {
+            thread = Thread.ofPlatform().name(THREAD_NAME).start(() -> { });
+        }
+        thread.join();
+        done();
+    }
+}
+
+public class SuspendAfterDeath extends TestScaffold {
+    private volatile ThreadReference thread;
+    private volatile boolean breakpointReached = false;
```

Comment:
The volatile-write to initialise this to false is not needed here.

Reply:
Actually I was a bit confused as to why thread was declared volatile, so I just followed the pattern. Maybe you can explain since you wrote it.

Reply:
II'm just pointing out that initialising breakpointReached to false is unnecessary as it's the default value.

Reply:
but do either of these fields need to be volatile?